torchtext/experimental/datasets/raw/question_answer.py

```diff
@@ -0,0 +1,78 @@
+import torch
+from torchtext.utils import download_from_url
+import json
+
+URLS = {
+    'SQuAD1':
+        ['https://rajpurkar.github.io/SQuAD-explorer/dataset/train-v1.1.json',
+         'https://rajpurkar.github.io/SQuAD-explorer/dataset/dev-v1.1.json'],
+    'SQuAD2':
+        ['https://rajpurkar.github.io/SQuAD-explorer/dataset/train-v2.0.json',
+         'https://rajpurkar.github.io/SQuAD-explorer/dataset/dev-v2.0.json']
+}
+
+
+def _create_data_from_json(data_path):
+    with open(data_path) as json_file:
+        raw_json_data = json.load(json_file)['data']
+        for layer1 in raw_json_data:
+            for layer2 in layer1['paragraphs']:
+                for layer3 in layer2['qas']:
+                    processed = {'context': layer2['context'], 'question': layer3['question'],
+                                 'answers': [item['text'] for item in layer3['answers']],
+                                 'answer_start': [item['answer_start'] for item in layer3['answers']]}
+                    if len(processed['answers']) == 0:
+                        processed['answers'] = [""]
+                        processed['answer_start'] = [-1]
+                    yield processed
+
+
+class RawQuestionAnswerDataset(torch.utils.data.IterableDataset):
+    """Defines an abstraction for raw question answer iterable datasets.
+    """
+
+    def __init__(self, iterator, start=0, num_lines=None):
```

Comment:
The other dataset still has the setup_iter method, so I'd mirror that behavior or change the behavior of the other.

Reply:
OK, let's keep this consistent for now since there are several ongoing PRs to re-write datasets. Once done, we can send out a separate PR to fix it.